io-engine/src/bin/io-engine-client/v1/stats_cli.rs

```diff
@@ -235,6 +244,93 @@ async fn nexus(mut ctx: Context, matches: &ArgMatches) -> crate::Result<()> {
     Ok(())
 }
 
+async fn replica(mut ctx: Context, matches: &ArgMatches) -> crate::Result<()> {
+    ctx.v2("Requesting Replica metrics");
+    let replica_name = matches.get_one::<String>("name");
+    let response = ctx
+        .v1
+        .stats
+        .get_replica_io_stats(v1rpc::stats::ListStatsOption {
+            name: replica_name.cloned(),
+        })
+        .await
+        .context(GrpcStatus)?;
+    match ctx.output {
+        OutputFormat::Json => {
+            println!(
+                "{}",
+                serde_json::to_string_pretty(response.get_ref())
+                    .unwrap()
+                    .to_colored_json_auto()
+                    .unwrap()
+            );
+        }
+        OutputFormat::Default => {
+            let stats: &Vec<v1rpc::stats::ReplicaIoStats> =
+                &response.get_ref().stats;
+            if stats.is_empty() {
+                if let Some(name) = replica_name {
+                    ctx.v1(&format!(
+                        "No IoStats found for {}, Check if device exists",
+                        name
+                    ));
+                } else {
+                    ctx.v1("No Replica IoStats found");
+                }
+                return Ok(());
+            }
+
+            let ticks_time = |ticks, tick_rate| -> String {
+                ticks_to_time(ticks, tick_rate).to_string()
+            };
+
+            let table = stats
+                .iter()
+                .map(|p| {
+                    let io_stat = p.stats.as_ref().unwrap();
+                    let tick_rate = io_stat.tick_rate;
+                    vec![
+                        io_stat.name.clone(),
+                        io_stat.num_read_ops.to_string(),
+                        adjust_bytes(io_stat.bytes_read),
+                        io_stat.num_write_ops.to_string(),
+                        adjust_bytes(io_stat.bytes_written),
+                        io_stat.num_unmap_ops.to_string(),
+                        adjust_bytes(io_stat.bytes_unmapped),
+                        ticks_time(io_stat.read_latency_ticks, tick_rate),
+                        ticks_time(io_stat.write_latency_ticks, tick_rate),
+                        ticks_time(io_stat.unmap_latency_ticks, tick_rate),
+                        ticks_time(io_stat.max_read_latency_ticks, tick_rate),
+                        ticks_time(io_stat.min_read_latency_ticks, tick_rate),
+                        ticks_time(io_stat.max_write_latency_ticks, tick_rate),
+                        ticks_time(io_stat.min_write_latency_ticks, tick_rate),
```

Comment:
As I mentioned you don't need to pass tick_rate, you can capture it in the closure:

```suggestion
            let table = stats
                .iter()
                .map(|p| {
                    let io_stat = p.stats.as_ref().unwrap();
                    let tick_rate = io_stat.tick_rate;
                    let ticks_time = |ticks| -> String {
                        ticks_to_time(ticks, tick_rate).to_string()
                    };
                    vec![
                        io_stat.name.clone(),
                        io_stat.num_read_ops.to_string(),
                        adjust_bytes(io_stat.bytes_read),
                        io_stat.num_write_ops.to_string(),
                        adjust_bytes(io_stat.bytes_written),
                        io_stat.num_unmap_ops.to_string(),
                        adjust_bytes(io_stat.bytes_unmapped),
                        ticks_time(io_stat.read_latency_ticks),
                        ticks_time(io_stat.write_latency_ticks),
                        ticks_time(io_stat.unmap_latency_ticks),
                        ticks_time(io_stat.max_read_latency_ticks),
                        ticks_time(io_stat.min_read_latency_ticks),
                        ticks_time(io_stat.max_write_latency_ticks),
                        ticks_time(io_stat.min_write_latency_ticks),
```

Reply:
Incorporated